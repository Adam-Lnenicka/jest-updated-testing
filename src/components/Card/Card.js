import React from "react";
import Upvotest from "../Upvotes/Upvotes";

const Card = ({ name, image, desc, price }) => {
  return (
    <div className="card">
      <h2>{name}</h2>
      <img src={image} alt={name} />
      <p>{desc}</p>
      <h4>Price: Â£{price}</h4>
      <Upvotest />
    </div>
  );
};

export default Card;
