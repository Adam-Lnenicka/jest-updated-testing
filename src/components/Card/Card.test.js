import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";
import Card from "./Card";

describe("Card", () => {
  it("matches snapshot", () => {
    const props = {
      name: "name",
      image: "image",
      desc: "desc",
      price: "price",
    };
    const wrapper = shallow(<Card {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
