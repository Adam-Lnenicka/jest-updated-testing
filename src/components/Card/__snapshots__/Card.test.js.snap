// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
<div
  className="card"
>
  <h2>
    name
  </h2>
  <img
    alt="name"
    src="image"
  />
  <p>
    desc
  </p>
  <h4>
    Price: Â£
    price
  </h4>
  <Upvotest />
</div>
`;
